src/gluonts/time_feature/lag.py

```diff
@@ -51,6 +51,28 @@ def get_granularity(freq_str: str) -> Tuple[int, str]:
     return multiple, granularity
 
 
+# A dictionary mapping granularity to the period length of the longest season
+# one can expect given the granularity of the time series.
+# This is similar to the frequency value in the R forecast package:
+# https://stats.stackexchange.com/questions/120806/frequency-value-for-seconds-minutes-intervals-data-in-r
+# This is useful for setting default values for past/context length for models
+# that do not do data augmentation and uses a single training example per time series in the dataset.
+FREQ_LONGEST_PERIOD_DICT = {
+    "M": 12,  # yearly seasonality
+    "W": 52,  # yearly seasonality
+    "D": 365,  # yearly seasonality
+    "B": 365,  # yearly seasonality
+    "H": 168,  # weekly seasonality
+    "T": 1440,  # daily seasonality
+    "min": 1440,  # daily seasonality
+}
+
+
+def longest_period_from_frequency_str(freq_str: str) -> int:
+    multiple, granularity = get_granularity(freq_str)
+    return FREQ_LONGEST_PERIOD_DICT[granularity] // multiple
```

Comment:
This may be minor, but: this line will round down the period length. For example, with `freq_str="3W"` it will yield `52//3` which is 17, but to cover a year one would need to use 18.

Reply:
I mean, if a model looks over multiple periods, then this may be almost irrelevant; it's more relevant if one looks at a single period of course.

Reply:
yes, year has slightly more than 52 weeks. Since this is minor (only used to determine the training length) I went with 52 which is also the recommended frequency value in R forecast package.

Reply:
I think 52 is fine, I was referring to `52//3` rounding down to something shorter than 52 weeks.

But again, I think this is minor, feel free to resolve this conversation.